test/contquadratic.jl

```diff
@@ -0,0 +1,219 @@
+# Continuous quadratic problems
+
+function contquadratictest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+    @testset "Testing quadratic interface with $solver" begin
+        @testset "QP1 - Quadratic objective" begin
+            # simple quadratic objective
+            # Min x^2 + 0.5xy + y^2 + 0.5yz + z^2
+            # st  x + 2y + 3z >= 4 (c1)
+            #     x +  y      >= 1 (c2)
+            #     x,y \in R
+
+            @test MOI.supportsproblem(solver, MOI.ScalarQuadraticFunction, [(MOI.ScalarAffineFunction{Float64},MOI.GreaterThan)])
+
+            m = MOI.SolverInstance(solver)
+
+            v = MOI.addvariables!(m, 3)
+            @test MOI.getattribute(m, MOI.VariableCount()) == 3
+
+            c1 = MOI.addconstraint!(m, MOI.ScalarAffineFunction(v, [1.0,2.0,3.0], 0.0), MOI.GreaterThan(4.0))
+            @test MOI.getattribute(m, MOI.ConstraintCount()) == 1
+
+            c2 = MOI.addconstraint!(m, MOI.ScalarAffineFunction([v[1],v[2]], [1.0,1.0], 0.0), MOI.GreaterThan(1.0))
+            @test MOI.getattribute(m, MOI.ConstraintCount()) == 2
+
+            MOI.setobjective!(m, MOI.MinSense, MOI.ScalarQuadraticFunction(v, [0.0,0.0,0.0],[v[1], v[1], v[2], v[2], v[3]], [v[1], v[2], v[2], v[3], v[3]], [2.0, 1.0, 2.0, 1.0, 2.0], 0.0))
+            @test MOI.getattribute(m, MOI.Sense()) == MOI.MinSense
+
+            if MOI.cangetattribute(m, MOI.ObjectiveFunction())
+                obj = MPI.getattribute(m, MOI.ObjectiveFunction())
```

Comment:
We need to document the structure we expect for returned functions. It's probably not reasonable to expect the quadratic function to be returned in the exact same order as was given for input. We could possibly impose CSC format with increasing indices within a column, or don't impose anything and make comparisons after normalizing.